static/js/recent_topics.js

```diff
@@ -0,0 +1,79 @@
+var recent_topics = (function () {
+
+var exports = {};
+
+var topics = new Dict(); // Key is stream-id:subject.
+
+exports.process_all_messages = function () {
+    exports.process_messages(message_list.all.all_messages());
+};
+
+exports.process_messages = function (messages) {
+    topics = new Dict();
+    messages.forEach(exports.process_message);
+};
+
+exports.process_message = function (msg) {
+    var is_ours = msg.sender_id === page_params.user_id;
+    var is_relevant = is_ours && msg.type === 'stream';
+    var key = msg.stream_id + ':' + msg.subject;
+    var topic = topics.has(key) ? topics.get(key) : undefined;
+    if (topic === undefined && !is_relevant) {
+        return false;
+    }
+    if (!topic) {
+        topics.set(key, {
+            our_last_msg: msg,
```

Comment:
I would consider using message ids here, instead of `msg`, as the way to remember recent messages.  This way we don't have to worry about the `msg` data getting stale due to message edits, etc.  (Well, we still need to think about message edits, but in principle we can let the downstream code call something like `message_store.get(msg_id)` when we need actual content.

Reply:
I was using the complete message object for easily comparing the timestamp. In any case, I would have to make a new object to store both the ids and the timestamps for comparison.

I'll make sure that the downstream code calls `message_store.get()`, and thus, changing it to return message ids instead.